pkgs/by-name/nf/nfd/package.nix

```diff
@@ -0,0 +1,67 @@
+{
+  lib,
+  stdenv,
+  boost, # probably needs to match the one from ndn-cxx
+  fetchFromGitHub,
+  libpcap,
+  ndn-cxx,
+  openssl,
+  dpkg,
+  pkg-config,
+  sphinx,
+  systemd,
+  wafHook,
+  websocketpp,
+  withSystemd ? lib.meta.availableOn stdenv.hostPlatform systemd,
+  withWebSocket ? true,
+}:
+
+stdenv.mkDerivation rec {
+  pname = "nfd";
+  version = "24.07";
+
+  src = fetchFromGitHub {
+    owner = "named-data";
+    repo = lib.toUpper pname;
+    rev = "NFD-${version}";
+    hash = "sha256-iEI8iS0eLLVe6PkOiCHL3onYNVYVZ1ttmk/aWrBkDhg=";
+    fetchSubmodules = true;
```

Comment:
We use our websocketpp. Why do we need submodules then https://github.com/named-data/NFD/blob/master/.gitmodules ?

Reply:
IIRC they used a different older version of the library in previous releases and the submodule one would take precedence. It seem to work correctly now with the nixpkgs one.

Reply:
I managed to get it to use the lib from nixpkgs.